core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -541,6 +548,7 @@ public synchronized void checkpoint() throws IOException {
       mSnapshotAllowed.set(true);
       catchUp(mStateMachine, client);
       mStateMachine.takeLocalSnapshot();
+      mLastCheckPointTime = System.currentTimeMillis();
```

Comment:
This is the external checkpoint mechanism. If you to record snapshots, you'll need to move this into the `JournalStateMachine#takeSnapshot` function. That function isn't called anywhere internally (Alluxio code) but is called externally by the ratis library when the number of journal entries exceeds the Alluxio PK `MASTER_JOURNAL_CHECKPOINT_PERIOD_ENTRIES`

Reply:
i guess for the ticket I was working on, it was specifically looking for metrics on the checkpoint. maybe we also need it for snapshots?

Reply:
From my understanding of the terminology, checkpoint is the rough equivalent of snapshot? In this implementation, we would only get the metrics for when a user explicitly initiated the checkpoint (snapshot) with `alluxio fsadmin journal checkpoint` or something like that.

The master also takes automatic checkpoints (snapshots) at specified intervals determined by the Alluxio PK mentioned earlier. The snapshots initiated by the automatic threshold mechanism won't have the metrics recorded properly if we only record the timing here